src/providers/ipa/ipa_subdomains.c

```diff
@@ -1779,6 +1780,272 @@ static errno_t ipa_subdomains_view_name_recv(struct tevent_req *req)
+        goto immediately;
+    }
+
+    DEBUG(SSSDBG_TRACE_ALL,
+          "Searching for overrides in view [%s] with filter [%s].\n",
+          state->ipa_view_name, filter);
+
+    base = talloc_asprintf(state, "cn=%s,cn=views,cn=accounts,%s",
+                           state->ipa_view_name, ldap_basedn);
+    if (base == NULL) {
+        DEBUG(SSSDBG_OP_FAILURE, "talloc_asprintf failed.\n");
+        ret = ENOMEM;
+        goto immediately;
+    }
+
+    subreq = sdap_get_generic_send(state, ev, sd_ctx->sdap_id_ctx->opts,
+                                 sh, base,
+                                 LDAP_SCOPE_SUBTREE,
+                                 filter, attrs,
+                                 NULL,
+                                 0,
+                                 dp_opt_get_int(state->sd_ctx->sdap_id_ctx->opts->basic,
+                                                SDAP_SEARCH_TIMEOUT),
+                                 false);
+    if (subreq == NULL) {
+        DEBUG(SSSDBG_OP_FAILURE, "sdap_get_generic_send failed.\n");
+        ret = ENOMEM;
+        goto immediately;
+    }
+
+    tevent_req_set_callback(subreq, ipa_subdomains_view_template_done, req);
+
+    return req;
+
+immediately:
+    if (ret == EOK) {
+        tevent_req_done(req);
+    } else {
+        tevent_req_error(req, ret);
+    }
+    tevent_req_post(req, ev);
+
+    return req;
+}
+
+static void ipa_subdomains_view_template_done(struct tevent_req *subreq)
+{
+    struct ipa_subdomains_view_template_state *state;
+    struct tevent_req *req;
+    size_t reply_count;
+    struct sysdb_attrs **reply;
+    struct sss_domain_info *trusted_dom = NULL;
+    const char *homedir = NULL;
+    const char *shell = NULL;
+    const char *anchor = NULL;
+    char **anchor_sid = NULL;
+    int anchor_num;
+    bool domain_template;
+    bool global_template;
+    errno_t ret;
+
+    req = tevent_req_callback_data(subreq, struct tevent_req);
+    state = tevent_req_data(req, struct ipa_subdomains_view_template_state);
+
+    ret = sdap_get_generic_recv(subreq, state, &reply_count, &reply);
+    talloc_zfree(subreq);
+    if (ret != EOK) {
+        DEBUG(SSSDBG_OP_FAILURE, "ipa_subdomains_view_template request failed.\n");
+        goto done;
+    }
+
+    if (reply_count == 0) {
+        DEBUG(SSSDBG_TRACE_ALL, "No override template found.\n");
+        tevent_req_done(req);
+        return;
+    }
+
+    DEBUG(SSSDBG_TRACE_FUNC,
+          "Found [%zu] overrides\n",
+          reply_count);
+
+    for (int c = 0; c < reply_count; c++) {
+        homedir = NULL;
+        shell = NULL;
+
+        ret = sysdb_attrs_get_string(reply[c], SYSDB_HOMEDIR, &homedir);
+        if (ret != EOK && ret != ENOENT) {
+            DEBUG(SSSDBG_OP_FAILURE,
+                  "sysdb_attrs_get_string failed: [%d](%s)\n",
+                  ret, sss_strerror(ret));
+            goto done;
+        }
+
+        sysdb_attrs_get_string(reply[c], SYSDB_SHELL, &shell);
```

Comment:
```
CID 583159: (#1 of 1): Unchecked return value (CHECKED_RETURN)
5. check_return: Calling sysdb_attrs_get_string without checking return value (as is done elsewhere 123 out of 124 times). 
```

Not sure why Covscan didn't report this previous scan...

Reply:
Fixed.

Reply:
Okay I agree. Updated.